service/namespace.go

```diff
@@ -11,7 +14,12 @@ const resultChannel string = "Result"
 
 // namespace returns the namespace of the service.
 func (service *Service) namespace() []string {
-	return []string{service.Hash}
+	h, err := hex.DecodeString(service.Hash)
```

Comment:
why not just using `h := []byte(service.Hash)` since hash is not actually a hex encoded word